### PR TITLE
chore(docs): Add clarification about `fastify.setErrorHandler()`

### PR DESCRIPTION
**Pull request by codershiba**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Reference/Errors.md [hunk 1]
```diff
@@ -169,6 +169,71 @@ Some things to consider in your custom error handler:
     internally monitors the error invocation to avoid infinite loops for errors
     thrown in the reply phases of the lifecycle. (those after the route handler)
 
+*Note: When utilizing Fastify's custom error handling through
+[`setErrorHandler`](./Server.md#seterrorhandler),
+it's crucial to be aware of how errors 
+are propagated between custom and default error handlers.*
```
Comment (jsumners) [inline docs/Reference/Errors.md:175]:
```suggestion
**Important**: When utilizing Fastify's custom error handling through
[`setErrorHandler`](./Server.md#seterrorhandler),
it's crucial to be aware of how errors 
are propagated between custom and default error handlers.
```

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Review by metcoder95 — APPROVED** [review]

**Comment by Fdawgs** [comment]
Quite a few issues with this, was going to review it.

**Comment by mcollina** [comment]
Ouch, sorry about it. How would you prefer to proceed?

**Comment by codershiba** [comment]
@Fdawgs what were the issues? If needed, I can open a follow-up PR to address those issues.

**Comment by Fdawgs** [comment]
@codershiba 
- Commit/PR title should have had the [`docs` prefix](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
- Doesn't follow [Fastify's documentation style guide](https://github.com/fastify/fastify/blob/main/docs/Guides/Style-Guide.md); use of contractions are a no-no for formal docs
- Few weasel/filler words that aren't needed i.e. "it's crucial to be aware" could have been cut down to "be aware"
- The `**Important**:` bit isn't needed, and the additional paragraph should have just been integrated into the original text

**Comment by codershiba** [comment]
@Fdawgs created #5267 to fix the issues

**Pull request merged**